### PR TITLE
Separate tests that use XML binding (e.g. xml_binding) in com/sun/ts/tests/jaxrs/spec/provider/standard, com/sun/ts/tests/jaxrs/spec/client/typedentities #518

### PR DESCRIPTION
**Pull request by scottmarlow**
Signed-off-by: Scott Marlow <smarlow@redhat.com>

https://github.com/eclipse-ee4j/jakartaee-tck/issues/518

### CODE REVIEW

**Review by gurunrao — COMMENTED** [review]
For some of the new files copyright has been backdated. we can keep it 2020.

File: src/com/sun/ts/tests/jaxrs/spec/client/typedentities/JAXRSClient.java [hunk 1]
```diff
@@ -29,11 +29,9 @@
 import java.io.Reader;
 
 import jakarta.activation.DataSource;
-import jakarta.ws.rs.core.GenericType;
```
Comment (gurunrao) [inline src/com/sun/ts/tests/jaxrs/spec/client/typedentities/JAXRSClient.java:32]: does GenericType have dependency on xml_binding or its unused?

File: src/com/sun/ts/tests/jaxrs/spec/client/typedentitieswithxmlbinding/build.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+<project name="jaxrs_spec_client_typedentitieswithjaxb" basedir="." default="usage">
+
+	<property name="app.name" value="jaxrs_spec_client_typedentitieswithjaxb" />
+
+	<property name="resource.classes" value="com/sun/ts/tests/jaxrs/spec/client/typedentitieswithjaxb/Resource.class,
+		com/sun/ts/tests/jaxrs/spec/client/typedentitieswithjaxb/EntityMessageWriter.class,
+		com/sun/ts/tests/jaxrs/spec/client/typedentitieswithjaxb/EntityMessageReader.class,
+		com/sun/ts/tests/jaxrs/ee/rs/ext/messagebodyreaderwriter/ReadableWritableEntity.class" />
+	<property name="appconfig.class" value="com/sun/ts/tests/jaxrs/spec/client/typedentitieswithjaxb/TSAppConfig.class" />
```
Comment (alwin-joseph) [inline src/com/sun/ts/tests/jaxrs/spec/client/typedentitieswithxmlbinding/build.xml:27]: it should be com/sun/ts/tests/jaxrs/spec/client/typedentitieswithxmlbinding right? or change the path to com/sun/ts/tests/jaxrs/spec/client/typedentitieswithjaxb at other places.

**Review by alwin-joseph — CHANGES_REQUESTED** [review]

**Comment by scottmarlow** [comment]
> In src/com/sun/ts/tests/jaxrs/spec/client/typedentities/JAXRSClient.java:
> > @@ -29,11 +29,9 @@
>  import java.io.Reader;
>  
>  import jakarta.activation.DataSource;
> -import jakarta.ws.rs.core.GenericType;
> 
> does GenericType have dependency on xml_binding or its unused?

@gurunrao the GenericType dependency moved to the typedentitieswithxmlbinding/JAXRSClient.java:
```
cd jakartaee-tck/src/com/sun/ts/tests/jaxrs/spec/client/typedentitieswithxmlbinding
grep -ri GenericType
JAXRSClient.java:import jakarta.ws.rs.core.GenericType;
JAXRSClient.java:    GenericType<JAXBElement<String>> type = new GenericType<JAXBElement<String>>() {
```

File: src/com/sun/ts/tests/jaxrs/spec/client/typedentitieswithxmlbinding/build.xml (hunk 2 above)
Comment (scottmarlow) [reply src/com/sun/ts/tests/jaxrs/spec/client/typedentitieswithxmlbinding/build.xml]: Fixed in updated commit.

**Comment by scottmarlow** [comment]
> For some of the new files copyright has been backdated. we can keep it 2020.

Thanks @gurunrao!  This is now fixed.

**Review by gurunrao — APPROVED** [review]
LGTM

**Comment by scottmarlow** [comment]
I'm running a test locally:
```
podman run -it centos /bin/bash
yum install -y zip java-1.8.0-openjdk-devel wget vim git unzip
export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk
export PATH=/usr/lib/jvm/java-1.8.0-openjdk/bin:$PATH

wget https://downloads.apache.org/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.zip
unzip apache-maven-3.6.3-bin.zip
export MAVEN_HOME=$PWD/apache-maven-3.6.3
export MAVEN_OPTS=-Xmx1800m -XX:-UseGCOverheadLimit
export PATH=$MAVEN_HOME/bin:$PATH
echo using $MAVEN_HOME

wget https://archive.apache.org/dist/ant/binaries/apache-ant-1.10.6-bin.zip
unzip -o apache-ant-1.10.6-bin.zip; rm -f apache-ant-1.10.6-bin.zip
export ANT_HOME=/apache-ant-1.10.6
export PATH=$ANT_HOME/bin:$PATH
export ANT_OPTS="-Djavax.xml.accessExternalStylesheet=all -Djavax.xml.accessExternalSchema=all -Djavax.xml.accessExternalDTD=file,http -Duser.home=$HOME"
export PATH=$JAVA_HOME/bin:$ANT_HOME/bin/:$PATH
 
cd /tmp
mkdir standalonetck
cd standalonetck
git clone https://github.com/eclipse-ee4j/jakartaee-tck
cd jakartaee-tck
export WORKSPACE=$PWD
export CTS_HOME=$WORKSPACE
export GF_BUNDLE_URL=https://download.eclipse.org/ee4j/glassfish/glassfish-6.0.0-SNAPSHOT-nightly.zip

cd $WORKSPACE
git checkout -b scott_jaxrs_move_jaxbtests master
git pull --rebase https://github.com/scottmarlow/jakartaee-tck.git jaxrs_move_jaxbtests
sh docker/build_standalone-tcks.sh jaxrs 2>&1 | tee build_standalone-tcks.log

export KEYWORDS="xml_binding"
sh docker/jaxrstck.sh 2>&1 | tee run_jaxrs_binding.log
```

I will post the results when it completes.

**Comment by alwin-joseph** [comment]
FYI, I ran the changes for jaxrs tck here https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck-alw/job/pr%2F524/5/ and it seem to work as expected. Total 123 tests ran.

**Review by alwin-joseph — APPROVED** [review]

**Comment by scottmarlow** [comment]
> FYI, I ran the changes for jaxrs tck here https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck-alw/job/pr%2F524/5/ and it seem to work as expected. Total 123 tests ran.

Thanks!, then I will stop trying to run the tests locally (which seem to be hitting https://github.com/eclipse-ee4j/glassfish/issues/23191)...

**Review by RohitKumarJain — APPROVED** [review]
LGTM.

**Pull request merged**